devices/temp_nix.go

```diff
@@ -28,6 +32,35 @@ func getTemps(temps map[string]int) map[string]error {
 			temps[label] = int(sensor.Temperature)
 		}
 	}
+
+	block, err := ghw.Block()
+	if err != nil {
+		log.Print("error getting block device info")
+		return nil
+	}
+
+	for _, disk := range block.Disks {
+		dev, err := smart.Open("/dev/" + disk.Name)
+		if err != nil {
+			log.Println(err)
+			continue
+		}
+		defer dev.Close()
+
+		switch sm := dev.(type) {
+		case *smart.SataDevice:
+			data, _ := sm.ReadSMARTData()
+			for _, attr := range data.Attrs {
+				if attr.Id == 194 {
```

Comment:
with the latest changes in `ataattr` branch the `data.Attrs` becomes a map and you can make this part a bit nicer as

```go
if attr, ok := range data.Attrs[194]; ok {
    // process the temperature value
    temps[disk.Name+"_"+disk.Model] = int(attr.Value)
}
```

Reply:
this is definitely more elegant! refactored in [3c148a4](https://github.com/xxxserxxx/gotop/pull/221/commits/3c148a4d97d0f54f1d6f464ddc64b03fda1ad5fa)